api/client_test.go

```diff
@@ -1201,3 +1200,88 @@ func TestClientWithNamespace(t *testing.T) {
 		t.Fatalf("Expected original namespace: \"%s\", got \"%s\"", ogNS, client.Namespace())
 	}
 }
+
+func TestVaultProxy(t *testing.T) {
+	const VaultHttpProxy string = "VAULT_HTTP_PROXY"
+	const VaultProxyAddr string = "VAULT_PROXY_ADDR"
+	const NoProxy string = "NO_PROXY"
+
+	tests := map[string]struct {
+		name                  string
+		vaultHttpProxy        string
+		vaultProxyAddr        string
+		noProxy               string
+		requestUrl            string
+		checkProxyVarPriority bool
```

Comment:
Instead of a bool, would it make sense to have this be a string that's the expected value for proxyUrl (or omit that check if empty string)?